[quant] torch.max_pool1d

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45175 [quant] Adding max_pool1d deps
* **#45152 [quant] torch.max_pool1d**

Differential Revision: [D23846473](https://our.internmc.facebook.com/intern/diff/D23846473)

Commit:
[quant] torch.max_pool1d

[ghstack-poisoned]

Commit:
Update on "[quant] torch.max_pool1d"


Differential Revision: [D23846473](https://our.internmc.facebook.com/intern/diff/D23846473)

[ghstack-poisoned]

vkuzo commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -2158,6 +2158,11 @@
   dispatch:
     MkldnnCPU: mkldnn_max_pool3d
 
+- func: quantized_max_pool1d(Tensor self, int[1] kernel_size, int[1] stride=[], int[1] padding=0, int[1] dilation=1, bool ceil_mode=False) -> Tensor
```
just curious, can we dispatch from `max_pool1d` directly?

z-a-f replied:
I will factor out both 1d and 2d in later PRs if you don't mind -- it is giving me some errors when I do it now, need to look deeper

vkuzo commented on aten/src/ATen/native/quantized/cpu/qpool.cpp:
```diff
@@ -395,9 +400,37 @@ Tensor quantized_max_pool2d(
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    bool ceil_mode) {
+  // (C, L) -> (C, 1, L) => kSqueezeDim = 1
+  // (N, C, L) -> (N, C, 1, L) => kSqueezeDim = 2
```
just double checking, this works as expected across different memory formats, right?

z-a-f replied:
Can you elaborate -- are you asking if squeeze/unsqueeze works properly across the memory formats, or if this trick does? For the latter -- it does, that what they do for the FP pooling ops. As for the squeeze/unsqueeze -- not sure, need to check the ref

vkuzo replied:
yes, was just asking if this function will work as expected for various memory formats.  If it does - sg.

vkuzo commented on aten/src/ATen/native/MaxPooling.cpp:
```diff
@@ -97,6 +97,10 @@ Tensor max_pool1d(
     IntArrayRef padding,
     IntArrayRef dilation,
     bool ceil_mode) {
+  if (self.is_quantized()) {
```
I was wondering if we can either delete this or delete the `quantized_max_pool1d`?

z-a-f replied:
That's a good point -- I'll try

z-a-f replied:
Actually, I started getting weird errors about the derivative -- I will factor it out in later PRs, if you don't mind

Commit:
Update on "[quant] torch.max_pool1d"


Differential Revision: [D23846473](https://our.internmc.facebook.com/intern/diff/D23846473)

[ghstack-poisoned]

Commit:
Update on "[quant] torch.max_pool1d"


Differential Revision: [D23846473](https://our.internmc.facebook.com/intern/diff/D23846473)

[ghstack-poisoned]

jerryzh168 commented on torch/nn/quantized/functional.py:
```diff
@@ -360,6 +360,22 @@ def linear(input, weight, bias=None, scale=None, zero_point=None):
     _packed_params = torch.ops.quantized.linear_prepack(weight, bias)
     return torch.ops.quantized.linear(input, _packed_params, scale, zero_point)
 
+def max_pool1d(input, kernel_size, stride=None, padding=0, dilation=1,
```
do we need this? i feel user won't be directly using these things in general

z-a-f replied:
This is to support the similarity to the `nn.functional`. Should I remove it?

Commit:
Update on "[quant] torch.max_pool1d"


Differential Revision: [D23846473](https://our.internmc.facebook.com/intern/diff/D23846473)

[ghstack-poisoned]
